### PR TITLE
bugfix: support carriage return

### PR DESCRIPTION
**Pull request by kasiaMarek**
connected to: https://github.com/scalameta/metals/issues/5369

### CODE REVIEW

File: scalameta/inputs/shared/src/main/scala/scala/meta/internal/inputs/InternalInput.scala [hunk 1]
```diff
@@ -21,8 +23,14 @@ trait InternalInput {
     val buf = new mutable.ArrayBuffer[Int]
     buf += 0
     var i = 0
+    var lastIsCR = false
     while (i < chars.length) {
       if (chars(i) == '\n') buf += (i + 1)
+      else if (lastIsCR) buf += i
```
[the remark this answers is not in the document]
Comment (kasiaMarek) [reply scalameta/inputs/shared/src/main/scala/scala/meta/internal/inputs/InternalInput.scala:29]: Is it enough to say: ```we consider all `\n`, `\r\n` and `\r` as new line```? This is consistent with [most editors](https://en.wikipedia.org/wiki/Comparison_of_text_editors#Newline_support).

File: scalameta/inputs/shared/src/main/scala/scala/meta/internal/inputs/InternalInput.scala [hunk 2]
```diff
@@ -21,8 +23,15 @@ trait InternalInput {
...
     buf += 0
     var i = 0
+    var lastIsCR = false
     while (i < chars.length) {
+      // we consider all `\n`, `\r\n` and `\r` as new line
       if (chars(i) == '\n') buf += (i + 1)
+      else if (lastIsCR) buf += i
+
+      if (chars(i) == '\r') lastIsCR = true
+      else lastIsCR = false
```
Comment (kitbellew) [inline scalameta/inputs/shared/src/main/scala/scala/meta/internal/inputs/InternalInput.scala:33]:
```suggestion
      lastIsCR = chars(i) == '\r'
```

**Review by kitbellew — CHANGES_REQUESTED** [review]
which system generates `\r` without a `\n`? and why? it's definitely not windows and not unix.

scalafmt definitely does not handle this case either. Token.CR is not something anyone looks for.

should we change the underlying system?

**Review by tgodzik — APPROVED** [review]
LGTM from me! @kitbellew what do you think?

**Review by kitbellew — APPROVED** [review]

**Pull request merged**